### PR TITLE
Add value to text mapping support in table panel

### PR DESCRIPTION
**Pull request by infernix**
Issue for this is https://github.com/grafana/grafana/issues/7119. ~I had this added on top of https://github.com/grafana/grafana/pull/9455 before but wanted to separate it from the text-to-value feature, so here it is :)~

This code replicates the behaviour from a singlestat panel to a table panel where values and value ranges can be mapped to text inside a table column.

### CODE REVIEW

File: public/app/plugins/panel/table/renderer.ts
Comment (torkelo) [inline public/app/plugins/panel/table/renderer.ts]: The code above needs a lot of unit tests :) 
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/heatmap/specs/renderer_specs.ts

**Review by torkelo — DISMISSED** [review]
This seems to be a somewhat popular request and enable some interesting scenarios. With some unit tests I think this might be mergable.

**Comment by infernix** [comment]
@torkelo updated, tests pass; not sure what needs to be done for coverage?

File: public/app/plugins/panel/table/renderer.ts
Comment (infernix) [reply public/app/plugins/panel/table/renderer.ts]: Are tests sufficiently covered now?

**Comment by marefr** [comment]
Hey @infernix we still need to think more about this PR in conjunction with #9455 and #10594 since all of them are very similar. Once we now were we want to take these, I'll have a look at this, hopefully next week.

**Review by marefr — APPROVED** [review]
I've made some changes so that the solution is more similar to the singlestat value mappings solution. Will include this when merging

**Comment by marefr** [comment]
@infernix thank you for contributing to Grafana! 

I had to make some changes to make the solution more similar to the singlestat value mappings and did some refactoring. Due to this I rebased your commit and my commit before merging it to master (95a4d61). Your commit unfortunately got rewritten as 67f0382222afa69fc8bc4690a4ef3770ac9449c1 leaving the author unchanged (@aapanasevich-sc), but replaced you as committer with me. My apologies for this.

This is merged so I'm closing this PR manually.

**Comment by lordberre** [comment]
Was this support dropped in grafana 7.x table/singlestat panels? I can only see value -> text there, not the other way around.

**Pull request closed without merging**
